shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/JDBCDatabaseCommunicationEngine.java

```diff
@@ -129,8 +129,8 @@ public ResponseHeader execute() {
         refreshMetaData(executionContext);
         ExecuteResult executeResultSample = result.iterator().next();
         return executeResultSample instanceof QueryResult
-                ? processExecuteQuery(executionContext, result.stream().map(each -> (QueryResult) each).collect(Collectors.toList()), (QueryResult) executeResultSample)
-                : processExecuteUpdate(executionContext, result.stream().map(each -> (UpdateResult) each).collect(Collectors.toList()));
+                ? processExecuteQuery(executionContext, (List) result, (QueryResult) executeResultSample)
+                : processExecuteUpdate(executionContext, (List) result);
```

Comment:
Can we consider about change the result of `proxySQLExecutor.execute(executionContext)` as List?

Reply:
Sure.

Reply:
@terrymanu I've migrated some Collection to List. But I have to use raw type because the List in method's parameter accepts different generic types.